test/InterestRateModule.t.sol

```diff
@@ -0,0 +1,130 @@
+        uint256 expectedInterestRate = config.baseRate + ((config.lowSlope / 10 ** 5) * utilisation);
+
+        // Then: actualInterestRate should be equal to expectedInterestRate
+        assertEq(actualInterestRate, expectedInterestRate);
+    }
+
+    function testSuccess_calculateInterestRate_OverOptimalUtilisation(
+        uint8 utilisationShift,
+        uint8 baseRate_,
+        uint8 highSlope_,
+        uint8 lowSlope_
+    ) public {
+        // Given: utilisation is between 80 and 100, InterestRateConfiguration setted as config
+        vm.assume(utilisationShift < 0.2 * 10 ** 5);
+        vm.assume(highSlope_ > lowSlope_);
+
+        uint256 utilisation = 0.8 * 10 ** 5 + uint256(utilisationShift);
+
+        DataTypes.InterestRateConfiguration memory config = DataTypes.InterestRateConfiguration({
+            baseRate: baseRate_,
+            highSlope: highSlope_,
+            lowSlope: lowSlope_,
+            utilisationThreshold: 0.8 * 10 ** 5
+        });
+
+        // When: creator calls setInterestConfig with config
+        vm.startPrank(creator);
+        interest.setInterestConfig(config);
+
+        // And: actualInterestRate is _calculateInterestRate with utilisation
+        uint256 actualInterestRate = interest._calculateInterestRate(utilisation);
+        vm.stopPrank();
+
+        // And: lowSlopeInterest is utilisationThreshold multiplied by lowSlope, highSlopeInterest is utilisation minus utilisationThreshold multiplied by highSlope
+        uint256 lowSlopeInterest = config.utilisationThreshold * (config.lowSlope / 10 ** 5);
+        uint256 highSlopeInterest = (utilisation - config.utilisationThreshold) * (config.highSlope / 10 ** 5);
+
+        // And: expectedInterestRate is baseRate added to lowSlopeInterest added to highSlopeInterest
+        uint256 expectedInterestRate = config.baseRate + lowSlopeInterest + highSlopeInterest;
+
+        // Then: actualInterestRate should be equal to expectedInterestRate
+        assertEq(actualInterestRate, expectedInterestRate);
+    }
+
+    function testRevert_setInterestConfig_NonOwner(
+        address unprivilegedAddress,
+        uint8 baseRate_,
+        uint8 highSlope_,
+        uint8 lowSlope_,
+        uint8 utilisationThreshold_
+    ) public {
+        // Given: unprivilegedAddress is not creator, InterestRateConfiguration setted as config
+        vm.assume(unprivilegedAddress != creator);
+
+        DataTypes.InterestRateConfiguration memory config = DataTypes.InterestRateConfiguration({
+            baseRate: baseRate_,
+            highSlope: highSlope_,
+            lowSlope: lowSlope_,
+            utilisationThreshold: utilisationThreshold_
+        });
+
+        vm.startPrank(unprivilegedAddress);
+        // When: unprivilegedAddress calls setInterestConfig
+        // Then: setInterestConfig should revert with UNAUTHORIZED
+        vm.expectRevert("UNAUTHORIZED");
+        interest.setInterestConfig(config);
+        vm.stopPrank();
+    }
+}
```

Comment:
Where are the tests for _updateInterestRate?

Reply:
@alikonuk1 can you pick this up?